integration-tests/src/test/java/oracle/kubernetes/operator/utils/Domain.java

```diff
@@ -981,23 +981,42 @@ public void findServerPropertyChange(String changedProperty, String serverName)
   }
 
   /**
-   * verify domain server pods get restarted after the property change by kubectl apply -f
-   * newDomainYamlfileWithChangedProperty
+   * verify domain server pods get restarted after the property change by kubectl apply -f new
+   * domain yaml file with added/changed property
    *
-   * @param fileNameWithChangedProperty
-   * @throws Exception
+   * @param fileNameWithChangedProperty - the fragment of domain yaml file with new added property
+   *     change
+   * @throws Exception - IOException when file is copied or errors occurred if the tested server is
+   *     not restarted
    */
   public void testDomainServerPodRestart(String fileNameWithChangedProperty) throws Exception {
     logger.info("Inside testDomainServerPodRestart domainYamlWithChangedProperty");
 
-    // kubectl apply the supplied domain yaml file under resources dir with changed property
-    String yamlDir = BaseTest.getProjectRoot() + "/integration-tests/src/test/resources/";
-    TestUtils.kubectlapply(yamlDir + fileNameWithChangedProperty);
+    // copy the original domain.yaml to domain_new.yaml
+    TestUtils.copyFile(
+        BaseTest.getUserProjectsDir() + "/weblogic-domains/" + domainUid + "/domain.yaml",
+        BaseTest.getUserProjectsDir() + "/weblogic-domains/" + domainUid + "/domain_new.yaml");
+
+    // append the file with changed property to the end of domain_new.yaml
+    TestUtils.concatFile(
```

Comment:
You can use Files.write(path, bytes, StandardOpenOption.APPEND);. Take a look at appendToDomainYamlAndCreate in Domain.java, you can refactor that if you want.

Reply:
Done